src/libsyntax/ast.rs

```diff
@@ -986,7 +986,9 @@ pub enum ExprKind {
     /// A `match` block.
     Match(P<Expr>, Vec<Arm>),
     /// A closure (for example, `move |a, b, c| {a + b + c}`)
-    Closure(CaptureBy, P<FnDecl>, P<Block>),
+    ///
+    /// The final span is the span of the argument block `|...|`
+    Closure(CaptureBy, P<FnDecl>, P<Block>, Span),
```

Comment:
I don't expect this PR to get approved in time for the other two, so I hauled the commit out into a separate PR: https://github.com/rust-lang/rust/pull/33125